### PR TITLE
feat(stepper): new design

### PR DESCRIPTION
**Pull request by matthprost**
## Type

- Feature

#### What is expected?

New design for `Stepper` component.

#### The following changes where made:

1. Removed side borders for + and - buttons

2. Changed colors and borders

3. Updated story

## Relevant logs and/or screenshots

Basic:
Screenshot 2021-05-12 at 17 10 47

Input hovered:
Screenshot 2021-05-12 at 17 16 26

Input focused:
Screenshot 2021-05-12 at 17 11 08

Side button hover:
Screenshot 2021-05-12 at 17 16 20

Sizes:
Screenshot 2021-05-12 at 17 11 15

Disabled:
Screenshot 2021-05-12 at 17 11 20

### CODE REVIEW

File: src/components/Stepper/index.js [hunk 1]
```diff
@@ -116,21 +91,23 @@ const StyledInput = styled.input`
...
+  border: 1px solid ${({ theme }) => theme.colors.gray300};
+  border-radius: 4px;
   ${({ disabled, theme }) =>
     disabled &&
     `
     > ${StyledTouchable}, ${StyledInput}, ${StyledCenterTouchable} {
       ${disabledStyles({ disabled, theme })}
     }
-  `}
+  `};
```
Comment (chambo-e) [inline src/components/Stepper/index.js:110]: 💩 ?

File: src/components/Stepper/index.js [hunk 2]
```diff
@@ -22,80 +23,54 @@ const disabledStyles = ({ disabled, theme }) =>
...
   position: relative;
+  border-radius: 4px;
+  width: ${({ size }) => containerSizes[size] - 10}px;
 
-  :hover,
-  :focus {
-    > svg {
-      fill: ${({ disabled, theme }) => !disabled && theme.colors.primary};
-    }
+  > svg {
```
Comment (chambo-e) [inline src/components/Stepper/index.js:47]: Possible to target the Icon instead pls ?

File: src/components/Stepper/index.js [hunk 3]
```diff
@@ -22,80 +23,54 @@ const disabledStyles = ({ disabled, theme }) =>
...
+  height: calc(100% - 8px);
   align-items: center;
   outline: none;
   justify-content: center;
-  border: 1px solid ${({ theme }) => theme.colors.gray350};
-
-  border-left-width: 0;
-  border-right-width: 0;
+  border-radius: 4px;
+  :hover:not([disabled]):not(:focus) {
```
Comment (chambo-e) [inline src/components/Stepper/index.js:66]: :wat: 😂 

There is no easier selector ?

Comment (matthprost) [reply src/components/Stepper/index.js:66]: I can change it for `:hover:not([disabled], :focus)` if you prefer 😄

File: src/components/Stepper/index.js (hunk 2 above)
Comment (matthprost) [reply src/components/Stepper/index.js]: Should be better @chambo-e

File: src/components/Stepper/index.js (hunk 3 above)
Comment (chambo-e) [reply src/components/Stepper/index.js]: That's just slightly better but yeah 😄

File: src/components/Stepper/index.js [hunk 4]
```diff
@@ -22,80 +23,56 @@ const disabledStyles = ({ disabled, theme }) =>
...
-  large: '16px',
-  medium: '10px',
-  small: '4px',
+const iconSizes = {
+  large: 26,
+  medium: 24,
+  small: 22,
 }
 
+const StyledIcon = styled(Icon)``
```
Comment (DorianMaliszewski) [inline src/components/Stepper/index.js:37]: Just Icon will be sufficient ;)

File: src/components/Stepper/index.js [hunk 5]
```diff
@@ -255,10 +234,7 @@ const Stepper = ({
         disabled={isPlusDisabled}
         aria-label="Plus"
       >
-        <StyledSeparator size={size} style={{ left: 0 }}>
-          |
-        </StyledSeparator>
-        <Icon name="plus" size={size === 'large' ? 28 : 18} color="gray300" />
+        <StyledIcon name="plus" size={iconSizes[size]} />
```
Comment (DorianMaliszewski) [inline src/components/Stepper/index.js:237]: Why `minus` icon has not the same size prop ??

Comment (matthprost) [reply src/components/Stepper/index.js:237]: my bad I forgot to change minus icon too

File: src/components/Stepper/index.js [hunk 6]
```diff
@@ -204,6 +183,16 @@ const Stepper = ({
   const isMinusDisabled = inputValue <= minValue || disabled
   const isPlusDisabled = inputValue >= maxValue || disabled
 
+  const centerTouchableRef = useRef(null)
+
+  const [inputMaxWidth, setInputMaxWidth] = useState()
+
+  useEffect(() => {
```
Comment (chambo-e) [inline src/components/Stepper/index.js:190]: This was to correct which behaviour ?

Comment (matthprost) [reply src/components/Stepper/index.js]: Input width but I corrected it with another way thanks to @DorianMaliszewski

**Review by DorianMaliszewski — CHANGES_REQUESTED** [review]
```js
  const [inputValue, setInputValue] = useState(parseIntOr(value, minValue))
  useEffect(() => {
    if (onChange) {
      onChange(inputValue)
    }
  }, [inputValue, onChange])
```

This is not a good way to handle controlled AND uncontrolled components. IMO, the component can have a local value but : 
- when `value` change its should change the local one too. 
- Each call to setInputValue should be followed by `onChange?.(event)` which contain the event of type `ChangeEvent<InputHtmlElement>`

**Review by DorianMaliszewski — APPROVED** [review]

**Review by chambo-e — APPROVED** [review]

**Pull request merged**